functions/src/index.js

```diff
@@ -0,0 +1,45 @@
+const {onRequest} = require("firebase-functions/v2/https");
+
+const { createProdia } = require("prodia");
+
+const prodiaKey = process.env.NEXT_PUBLIC_PRODIA_API_KEY;
```

Comment:
Think this secret would be stored via the firebase function config as function execute server side so we could access front end env variables. @shortxmas is this true?

Reply:
I forgot about that part. Idk how the configuration works, but the OpenAI key used `defineString("OPENAI_API_KEY")`. Sorry, still getting used to how firebase works

Reply:
I've fixed the code to use `defineString()`, but I don't know how to actually set `PRODIA_API_KEY` for deployment. I think it should be the same as how you set `OPENAI_API_KEY`.